index.js

```diff
@@ -768,7 +768,12 @@ SendStream.prototype.setHeader = function setHeader(path, stat){
   }
 
   if (this._etag && !res.getHeader('ETag')) {
-    var val = etag(stat)
+    var val = null;
+    if(typeof this._etag == 'function') {
+        val = this._etag(stat);
```

Comment:
Custom ETags are expected to be added by the headers event, which is the purpose of the surrounding if statement. If the intention is to move that functionality into a specific option, please remove the existing custom ETag functionality or at lease deprecate it.